Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/FilterAdapter.cpp

```diff
@@ -0,0 +1,501 @@
+            {
+                rowPath.Push(pathEntry);
+            }
+        }
+        return rowPath;
+    }
+
+    ValueStringFilter::ValueStringFilter()
+        : RowFilterAdapter()
+    {
+    }
+
+    void ValueStringFilter::SetFilterString(QString filterString)
+    {
+        if (m_filterString != filterString)
+        {
+            m_filterString = std::move(filterString);
+            InvalidateFilter();
+            SetFilterActive(!m_filterString.isEmpty());
+        }
+    }
+
+    RowFilterAdapter::MatchInfoNode* ValueStringFilter::NewMatchInfoNode(MatchInfoNode* parentRow) const
+    {
+        return new StringMatchNode(parentRow);
+    }
+
+    void ValueStringFilter::CacheDomInfoForNode(const Dom::Value& domValue, MatchInfoNode* matchNode) const
+    {
+        auto actualNode = static_cast<StringMatchNode*>(matchNode);
+        const bool nodeIsRow = IsRow(domValue);
+        AZ_Assert(nodeIsRow, "Only row nodes should be cached by a RowFilterAdapter");
+        if (nodeIsRow)
+        {
+            actualNode->m_matchableDomTerms.clear();
+            for (auto childIter = domValue.ArrayBegin(), endIter = domValue.ArrayEnd(); childIter != endIter; ++childIter)
+            {
+                auto& currChild = *childIter;
+                if (currChild.IsNode())
+                {
+                    auto childName = currChild.GetNodeName();
+                    if (childName != Dpe::GetNodeName<Dpe::Nodes::Row>()) // don't cache child rows, they have they're own entries
+                    {
+                        static const Name valueName("Value");
+                        auto foundValue = currChild.FindMember(valueName);
+                        if (foundValue != currChild.MemberEnd())
+                        {
+                            actualNode->AddStringifyValue(foundValue->second);
+                        }
+
+                        if (m_includeDescriptions)
+                        {
+                            static const Name descriptionName("Description");
+                            auto foundDescription = currChild.FindMember(descriptionName);
+                            if (foundDescription != currChild.MemberEnd())
+                            {
+                                actualNode->AddStringifyValue(foundDescription->second);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    bool ValueStringFilter::MatchesFilter(MatchInfoNode* matchNode) const
+    {
+        auto actualNode = static_cast<StringMatchNode*>(matchNode);
+        return (m_filterString.isEmpty() || actualNode->m_matchableDomTerms.contains(m_filterString, Qt::CaseInsensitive));
+    }
+
+    void ValueStringFilter::StringMatchNode::AddStringifyValue(const Dom::Value& domValue)
+    {
+        QString stringifiedValue;
+
+        if (domValue.IsNull())
+        {
+            stringifiedValue = QStringLiteral("null");
```

Comment:
Not 100% sure we'll want this, but I think it's fine to leave it as I'm not sure we'll ever even have a null value field.

Reply:
fair, but it doesn't hurt. Resolving...